src/executer.ts

```diff
@@ -0,0 +1,59 @@
+import Command from './command'
+
+const executer: any = {
+  init: function (Agent: any) {
+    executer.agent = Agent
+    executer.commandQueue = []
+    executer.agents = []
+  }
+}
+
+executer.run = function () {
+  let cmd: Command[] = executer.parseCommand(arguments)
+  cmd = executer.commandQueue.concat(cmd)
+  for (executer.commandQueue = []; 0 < cmd.length;) {
+    if (executer.execute(cmd[0])) cmd.shift()
+  }
+  executer.commandQueue = executer.commandQueue.concat(cmd)
+}
+
+executer.parseCommand = function (): Command[] {
+  let commands: Command[] = []
+  for (let i = 0; i < arguments.length; i++) {
+    try {
+      let command = new Command(arguments[i])
+      if (command.callProvide) {
+        // TODO call Agent.provide after create
+        continue
```

Comment:
WIP. 
here will call provide function and `provide` Command **NOT** add to queue.